src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts

```diff
@@ -99,4 +153,54 @@ export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBand
   setHasBandwidthPriority(hasBandwidthPriority: boolean): void {
     this.hasBandwidthPriority = hasBandwidthPriority;
   }
+
+  setTransceiverController(transceiverController: TransceiverController): void {
+    this.transceiverController = transceiverController;
+  }
+
+  async updateTransceiverController(): Promise<void> {
+    const settings = this.getStreamCaptureSetting();
+    if (!settings) {
+      return;
+    }
+
+    const encodingParams: RTCRtpEncodingParameters = this.calculateEncodingParameters(settings);
+    if (
+      !this.isEncodingParameterEqual(
+        encodingParams,
+        this.encodingParamMap.get(NScaleVideoUplinkBandwidthPolicy.encondingMapKey)
+      )
+    ) {
+      this.encodingParamMap.set(NScaleVideoUplinkBandwidthPolicy.encondingMapKey, encodingParams);
+      this.transceiverController.setEncodingParameters(this.encodingParamMap);
+    }
+  }
+
+  private isEncodingParameterEqual(
+    encoding1: RTCRtpEncodingParameters,
+    encoding2: RTCRtpEncodingParameters
+  ): boolean {
+    return (
+      encoding1.maxBitrate === encoding2.maxBitrate &&
+      encoding1.scaleResolutionDownBy === encoding2.scaleResolutionDownBy
+    );
+  }
+
+  private calculateEncodingParameters(setting: MediaTrackSettings): RTCRtpEncodingParameters {
+    const maxBitRates = this.maxBandwidthKbps() * 1000;
+    let scale = 1;
+    if (setting && this.scaleResolution && !this.hasBandwidthPriority && this.numParticipants > 2) {
+      const targetHeight = NScaleVideoUplinkBandwidthPolicy.targetHeightArray[this.numParticipants];
+
+      scale = Math.max(setting.height / targetHeight, 1);
+    }
+    return {
+      scaleResolutionDownBy: scale,
+      maxBitrate: maxBitRates,
+    };
+  }
+
+  private getStreamCaptureSetting(): MediaTrackSettings {
```

Comment:
`| undefined`.

Reply:
Is it to be explicit? TS does not complain so I did not do it :p

Reply:
It'll complain if we ever manage to turn on strict mode. Be explicit.

Reply:
Got it. Updated.